app/backend/src/couchers/i18n/i18n.py

```diff
@@ -3,132 +3,62 @@
+def get_i18next() -> I18Next:
     """
     Load all translation files from the locales directory and apply fallbacks.
-    Each locale JSON file contains components as top-level keys that map to
-    dictionaries of string translations. Fallbacks are prebaked so every language
-    has complete coverage using English as the base and applying fallbacks in the
-    correct precedence order.
 
     Returns:
-        Dictionary structure: lang -> component -> (string -> translated string)
+        An I18Next object that can localize the application strings.
 
     The result is cached so that translations are only loaded once per process.
     """
-    all_langs_all_strings: dict[str, dict[str, dict[str, str]]] = {}
 
-    locales_dir = Path(__file__).parent / "locales"
+    i18next = I18Next()
 
     # Load all locale JSON files from the locales directory
+    locales_dir = Path(__file__).parent / "locales"
     for locale_file in locales_dir.glob("*.json"):
-        lang = locale_file.stem  # e.g., "en" from "en.json"
+        lang_code = locale_file.stem  # e.g., "en" from "en.json"
 
         with open(locale_file, "r", encoding="utf-8") as f:
             translations = json.load(f)
 
-        # Initialize the language dictionary if needed
-        if lang not in all_langs_all_strings:
-            all_langs_all_strings[lang] = {}
+        plural_rule = PluralRules.for_language(lang_code) or PluralRules.en
+        language = i18next.add_language(lang_code, plural_rule)
+        language.load_json_dict(translations)
 
-        # Each top-level key in the JSON file is a component
-        for component_name, component_translations in translations.items():
-            if component_name not in all_langs_all_strings[lang]:
-                all_langs_all_strings[lang][component_name] = {}
-
-            # Store the translations for this component
-            all_langs_all_strings[lang][component_name] = component_translations
-
-    # Apply fallbacks: English is our source of truth - must exist
-    if "en" not in all_langs_all_strings:
+    # Apply ultimate fallback: English is our source of truth - must exist
+    en = i18next.languages_by_code.get("en")
+    if en is None:
         raise RuntimeError("English translations must be loaded")
+    i18next.fallback_language = en
 
-    en_strings = all_langs_all_strings["en"]
-
-    # Get all languages we need to process (loaded languages + those in fallback config)
-    all_languages = set(all_langs_all_strings.keys()) | set(LANGUAGE_FALLBACKS.keys())
-
-    for lang in all_languages:
-        if lang == "en":
-            continue  # English is already complete
+    # Apply other fallbacks
+    for from_lang_code, to_lang_code in LANGUAGE_FALLBACKS.items():
+        from_lang = i18next.languages_by_code.get(from_lang_code)
+        to_lang = i18next.languages_by_code.get(to_lang_code)
+        if from_lang is not None and to_lang is not None:
+            from_lang.fallback = to_lang.fallback
```

Comment:
I think there's a bug here, languages should form a linked list, not just short-circuit to the last fallback. If so, please add a unit test that catches this bug so it won't be re-introduced in the future.

```suggestion
            from_lang.fallback = to_lang
```

Reply:
Good catch. I added a failing test, then fixed the implementation.